data-browser/src/views/ErrorPage.tsx

```diff
@@ -60,3 +61,42 @@ function ErrorPage({ resource }: ResourcePageProps): JSX.Element {
 }
 
 export default ErrorPage;
+
+interface ErrorBoundaryProps {
+  children: React.ReactNode;
+  FallBackComponent: React.ComponentType;
+}
+
+interface ErrorBoundaryState {
+  error?: Error;
+}
+
+export class ErrorBoundary extends React.Component<
+  ErrorBoundaryProps,
+  ErrorBoundaryState
+> {
+  public constructor(props: ErrorBoundaryProps) {
+    super(props);
+    this.state = { error: undefined };
+  }
+
+  public static getDerivedStateFromError(error) {
+    // Update state so the next render will show the fallback UI.
+    return { error };
+  }
+
+  public render() {
+    if (this.state.error) {
+      // You can render any custom fallback UI    w
+      return (
+        <CrashPage
+          error={this.state.error}
+          clearError={() => this.setState(() => ({ error: undefined }))}
```

Comment:
Like useState the function argument version of setState is usually only for when you need to determine the state based on the previous state, meaning `setState({ error: undefined })` is enough in this case.